### PR TITLE
Update E0432 long description with the separate behaviors of editions 2015 and 2018

### LINKED ISSUE #64668: Outdated `--explain` text for E0432 "An import was unresolved."
See [E0432][1] for the current text.

[1]: https://doc.rust-lang.org/error-index.html#E0432

It currently includes:
«
Paths in `use` statements are relative to the crate root. To import items relative to the current and parent modules, use the `self::` and `super::` prefixes, respectively.
»
The first sentence is untrue in Rust 2018, and so the advice about `self::` is unhelpful.

It also says
«
Or, if you tried to use a module from an external crate, you may have missed the `extern crate` declaration (which is usually placed in the crate root):
»
which I think is unhelpful in Rust 2018.

### PR DESCRIPTION
**Pull request by poliorcetics**
Fixes #64668.

I restarted from the work done in #71413.

### CODE REVIEW

File: src/librustc_error_codes/error_codes/E0432.md [hunk 1]
```diff
@@ -12,21 +12,59 @@ prefixes, respectively. Also verify that you didn't misspell the import
 name and that the import exists in the module from where you tried to
 import it. Example:
 
+In Rust 2015, paths in `use` statements are relative to the crate root. To
+import items relative to the current and parent modules, use the `self::` and
+`super::` prefixes, respectively.
+
+<<<<<<< Updated upstream
```
Comment (bugadani) [inline src/librustc_error_codes/error_codes/E0432.md:19]: Is this intentional?

Comment (poliorcetics) [reply src/librustc_error_codes/error_codes/E0432.md:19]: No, that's a fail on my side. I'll remove it and squash

**Comment by GuillaumeGomez** [comment]
Thanks!

@bors: r+ rollup

**Pull request merged**